overlord/snapstate/handlers_components.go

```diff
@@ -724,6 +724,42 @@ func (m *SnapManager) doPrepareKernelModulesComponents(t *state.Task, _ *tomb.To
 	// Inject fault during the kernel components setup
 	osutil.MaybeInjectFault("prepare-kernel-components")
 
+	var newInfo *snap.Info
+	// Set the default to false for compatibility with older snapd (case of
+	// joint refresh of snapd and kernel).
+	prepareRestart := false
+	if err := t.Get("prepare-restart", &prepareRestart); err != nil &&
```

Comment:
isn't prepare-restart doubling what the restart-boundary should also indicates? I understand that is trickier for the finish-restart bit as the task is not the boundary one